clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsOnDemandCacheUpdater.groovy

```diff
@@ -126,4 +143,17 @@ class CatsOnDemandCacheUpdater implements OnDemandCacheUpdater {
       it.pendingOnDemandRequests(providerCache)
     }.flatten()
   }
+
+  @Override
+  Map pendingOnDemandRequest(OnDemandAgent.OnDemandType type, String cloudProvider, String id) {
```

Comment:
Not sure how we feel about returning `null` here or `Optional`.

I did it originally with `Optional` but it seemed a little clumsy to assemble everything downstream. 

Happy to revisit.

Reply:
null is fine IMO but I'd add `@Nullable`

Reply:
👍